src/views/List.tsx

```diff
@@ -1,21 +1,35 @@
-import React from 'react';
-import { ListItem as ListItemComponent } from '../components/ListItem';
-import { ListItem } from '../api';
+import { useState } from "react";
+import { ListItem as ListItemComponent } from "../components/ListItem";
+import { FilterList as FilterListComponent } from "../components/FilterList";
+import { ListItem } from "../api";
 
 interface Props {
 	data: ListItem[];
 }
 
 export function List({ data }: Props) {
-	const hasItem = data.length !== 0;
+	const [searchTerm, setSearchTerm] = useState<string>("");
+
+	const filteredData = data.filter((item) =>
```

Comment:
We should wrap this in a `useMemo`, which will cache a calculation between re-renders. That way if this component gets re-rendered but `searchTerm` hasn't changed between renders, we don't have to re-filter to list. It can be a big win for performance, especially as this list gets longer

https://react.dev/reference/react/useMemo

Reply:
That is a great call out. I had not used that hook before, so this gave me a chance to have a go at it.